### PR TITLE
[ONNX] Add dim argument to all symbolic

### LINKED ISSUE #65817: torch.all with dim Onnx: _all() takes 2 positional arguments but 4 were given
## 🐛 Bug

When I use the torch.all function with the dim parameter I get the following error during the Onnx translation: _all() takes 2 positional arguments but 4 were given 

## To Reproduce

```
import torch
from torch import nn

class TorchAll(nn.Module):
    def forward(self, tensor):
        tensor = torch.all(tensor, dim=1)
        return tensor

X = torch.ones((3, 300, 300), dtype=torch.int32)

torch.onnx.export(
    TorchAll(),
    (X), # Dummy input for shape
    "torch_all_model.onnx",
    opset_version=12,
    do_constant_folding=True,
)
```

## Expected behavior

Expecting the translation to occur without error

## Environment

PyTorch version: 1.8.1
Is debug build: False
CUDA used to build PyTorch: 10.1
ROCM used to build PyTorch: N/A

OS: Ubuntu 18.04.6 LTS (x86_64)
GCC version: (Ubuntu 7.5.0-3ubuntu1~18.04) 7.5.0
Clang version: Could not collect
CMake version: version 3.18.4
Libc version: glibc-2.27

Python version: 3.8.11 (default, Aug  3 2021, 15:09:35)  [GCC 7.5.0] (64-bit runtime)
Python platform: Linux-4.15.0-143-generic-x86_64-with-glibc2.17
Is CUDA available: True
CUDA runtime version: Could not collect
GPU models and configuration: 
GPU 0: GeForce GTX 1070
GPU 1: GeForce GTX 1070

Nvidia driver version: 450.119.03
cuDNN version: /usr/lib/x86_64-linux-gnu/libcudnn.so.7.6.5
HIP runtime version: N/A
MIOpen runtime version: N/A

Versions of relevant libraries:
[pip3] numpy==1.20.3
[pip3] torch==1.8.1
[pip3] torchaudio==0.8.0a0+e4e171a
[pip3] torchvision==0.9.1
[conda] blas                      1.0                         mkl  
[conda] cudatoolkit               10.1.243             h6bb024c_0  
[conda] ffmpeg                    4.3                  hf484d3e_0    pytorch
[conda] mkl                       2021.3.0           h06a4308_520  
[conda] mkl-service               2.4.0            py38h7f8727e_0  
[conda] mkl_fft                   1.3.0            py38h42c9631_2  
[conda] mkl_random                1.2.2            py38h51133e4_0  
[conda] numpy                     1.20.3           py38hf144106_0  
[conda] numpy-base                1.20.3           py38h74d4b33_0  
[conda] pytorch                   1.8.1           py3.8_cuda10.1_cudnn7.6.3_0    pytorch
[conda] torchaudio                0.8.1                      py38    pytorch
[conda] torchvision               0.9.1                py38_cu101    pytorch

## Additional context

Same error occur on my mac with PyTorch 1.9.1

### PR DESCRIPTION
**Pull request by shubhambhokare1**
Fixes #65817

### CODE REVIEW

File: test/onnx/test_pytorch_onnx_onnxruntime.py [hunk 1]
```diff
@@ -7434,6 +7434,13 @@ def forward(self, x):
         x = torch.tensor([[True, False], [False, False]])
         self.run_test(M(), (x, ))
 
+        class MDim(torch.nn.Module):
+            def forward(self, x):
+                return x.all(dim=1)
```
Comment (BowenBao) [inline test/onnx/test_pytorch_onnx_onnxruntime.py:7439]: nit: do we need to test the two options of `keepdim` too?

File: torch/onnx/symbolic_opset9.py [hunk 2]
```diff
@@ -2541,8 +2541,22 @@ def _any(g, input):
     input_sum = sym_help._reducesum_helper(g, input, keepdims_i=0)
     return gt(g, input_sum, g.op("Constant", value_t=torch.LongTensor([0])))
 
-def _all(g, input):
-    return g.op("Not", _any(g, g.op("Not", input)))
+def _all(g, *args):
+    # aten::all(Tensor self)
+    if len(args) == 1:
+        return g.op("Not", _any(g, g.op("Not", args[0])))
```
Comment (BowenBao) [inline torch/onnx/symbolic_opset9.py:2547]: nit: update _any and _all to have one reuse the other and both benefit from this update.

Comment (shubhambhokare1) [reply torch/onnx/symbolic_opset9.py]: Good catch, added that

File: test/onnx/test_pytorch_onnx_onnxruntime.py (hunk 1 above)
Comment (shubhambhokare1) [reply test/onnx/test_pytorch_onnx_onnxruntime.py:7453]: Added tests

**Review by BowenBao — APPROVED** [review]
